map.go

```diff
@@ -1506,17 +1506,42 @@ func newMapIterator(target *Map) *MapIterator {
 	}
 }
 
-// Next decodes the next key and value.
-//
-// Iterating a hash map from which keys are being deleted is not
-// safe. You may see the same key multiple times. Iteration may
-// also abort with an error, see IsIterationAborted.
+// nextQueueMap decodes the next value from a map of type
+// Queue or Stack.
 //
 // Returns false if there are no more entries. You must check
 // the result of Err afterwards.
+func (mi *MapIterator) nextQueueMap(valueOut interface{}) bool {
+	if mi.err != nil || mi.done {
+		return false
+	}
+
+	// For Queue/Stack map block the iteration after maxEntries
+	// to avoid potential infinite loops
+	// (values can be pushed to map while doing pop)
+	if mi.count == mi.maxEntries {
+		mi.err = fmt.Errorf("%w", ErrIterationAborted)
+		return false
+	}
+
+	mi.count++
```

Comment:
I don't think this check works as intended. If a user indeed uses a stack or queue as kernel to userspace mechanism, and is constantly adding new objects, then you would expect to iterate for more than the maxEntries of the map.

I think we can just leave this out.

Reply:
I see. In that way, is the user that decides whether to stop after certain iterations or not.